File: ruby/lib/verbs/phone_number.rb
require_relative 'xml_verb'

module RubyBandwidthBxml
  class PhoneNumber
    include XmlVerb
      def to_xml(xml)
        xml.PhoneNumber(number, compact_hash({
         'transferAnswerUrl' => transfer_answer_url,
         'transferAnswerMethod' => transfer_answer_method,
         'username' => username,
         'password' => password,
         'tag' => tag
       }))
    end
  end
end

File: java/src/main/java/com/bandwidth/sdk/voice/models/verbs/PhoneNumber.java
package com.bandwidth.sdk.voice.models.verbs;

import lombok.Builder;

import java.net.URI;

import javax.xml.bind.annotation.XmlAttribute;
import javax.xml.bind.annotation.XmlType;
import javax.xml.bind.annotation.XmlValue;

@XmlType(name = PhoneNumber.TYPE_NAME) @Builder
public class PhoneNumber {
    public static final String TYPE_NAME = "PhoneNumber";

    @XmlValue
    private String value;

    @XmlAttribute
    private URI transferAnswerUrl;

    @XmlAttribute
    private Method transferAnswerMethod;

    @XmlAttribute
    protected String username;

    @XmlAttribute
    protected String password;

    @XmlAttribute
    private String tag;

    public static class PhoneNumberBuilder{

        public PhoneNumberBuilder transferAnswerUrl(String uri){ return this.transferAnswerUrl(URI.create(uri)); }

        public PhoneNumberBuilder transferAnswerUrl(URI uri){
            this.transferAnswerUrl = uri;
            return this;
        }

        public PhoneNumberBuilder transferAnswerMethod(String method){ return this.transferAnswerMethod(Method.fromValue(method)); }

        public PhoneNumberBuilder transferAnswerMethod(Method method){
            this.transferAnswerMethod = method;
            return this;
        }

    }

}
